<?xml version="1.0" encoding="utf-8"?>

<launch>
  <!-- Console launch prefix -->
  <arg name="launch_prefix" default=""/>

  <!-- Config and weights folder. -->
  <arg name="yolo_weights_path"          default="$(find prometheus_detection)/src/darknet_ros/yolo_network_config/weights"/>
  <arg name="yolo_config_path"           default="$(find prometheus_detection)/src/darknet_ros/yolo_network_config/cfg"/>

  <!-- Load parameters -->
  <rosparam command="load" ns="darknet_ros" file="$(find prometheus_detection)/src/darknet_ros/config/ros.yaml"/>
  <rosparam command="load" ns="darknet_ros" file="$(find prometheus_detection)/src/darknet_ros/config/yolov3-tiny.yaml"/>

  <!-- Start darknet and ros wrapper -->
  <node pkg="prometheus_detection" type="darknet_ros" name="darknet_ros" output="screen" launch-prefix="$(arg launch_prefix)">
    <param name="weights_path"          value="$(arg yolo_weights_path)" />
    <param name="config_path"           value="$(arg yolo_config_path)" />
    <rosparam command="load" file="$(find prometheus_experiment)/config/prometheus_detection_config/camera_param.yaml" />
  </node>

 <!--<node name="republish" type="republish" pkg="image_transport" output="screen" 	args="compressed in:=/front_camera/image_raw raw out:=/camera/image_raw" /> -->
</launch>
